src/Plugins/RestServer/Controllers/v1/ContractsController.cs

```diff
@@ -0,0 +1,220 @@
+        public IActionResult GetContractManifest(
+            [FromRoute(Name = "hash")]
+            UInt160 scripthash)
+        {
+            var contracts = NativeContract.ContractManagement.GetContract(_neosystem.StoreView, scripthash);
+            if (contracts == null)
+                throw new ContractNotFoundException(scripthash);
+            return Ok(contracts.Manifest);
+        }
+
+        /// <summary>
+        /// Get nef of a smart contract.
+        /// </summary>
+        /// <param name="scripthash" example="0xed7cc6f5f2dd842d384f254bc0c2d58fb69a4761">ScriptHash</param>
+        /// <returns>Contract Nef object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("{hash:required}/nef", Name = "GetContractNefFile")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(NefFile))]
+        public IActionResult GetContractNef(
+            [FromRoute(Name = "hash")]
+            UInt160 scripthash)
+        {
+            var contracts = NativeContract.ContractManagement.GetContract(_neosystem.StoreView, scripthash);
+            if (contracts == null)
+                throw new ContractNotFoundException(scripthash);
+            return Ok(contracts.Nef);
+        }
+
+        /// <summary>
+        /// Invoke a method as ReadOnly Flag on a smart contract.
+        /// </summary>
+        /// <param name="scripthash" example="0xed7cc6f5f2dd842d384f254bc0c2d58fb69a4761">ScriptHash</param>
+        /// <param name="method" example="balanceOf">method name</param>
+        /// <param name="contractParameters">JArray of the contract parameters.</param>
+        /// <returns>Execution Engine object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpPost("{hash:required}/invoke", Name = "InvokeContractMethod")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ExecutionEngineModel))]
+        public IActionResult InvokeContract(
+            [FromRoute(Name = "hash")]
+            UInt160 scripthash,
+            [FromQuery(Name = "method")]
+            string method,
+            [FromBody]
+            ContractParameter[] contractParameters)
+        {
+            var contracts = NativeContract.ContractManagement.GetContract(_neosystem.StoreView, scripthash);
+            if (contracts == null)
+                throw new ContractNotFoundException(scripthash);
+            if (string.IsNullOrEmpty(method))
+                throw new QueryParameterNotFoundException(nameof(method));
+            try
+            {
+                var engine = ScriptHelper.InvokeMethod(_neosystem.Settings, _neosystem.StoreView, contracts.Hash, method, contractParameters, out var script);
```

Comment:
Need supporting signer account and scope like what invokefunction did, or methods can't pass verifywitness.

Reply:
![image](https://github.com/user-attachments/assets/cf1153ee-5fb9-40a8-b1e8-ae7189c2f4bd)